### ISSUE TITLE
508-defect-2 [SCREENREADER, COGNITION]: Find Form Results - Spanish content within the "Go to tool" link MUST be read in Spanish by a screen reader

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#23641]
[508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Josh

## User Story or Problem Statement

As a Spanish speaking screen reader user, I want Spanish content to be read to me in a Spanish accent.

## Details

Within the "Go to tool" link (styled as a button), the specific name of the form is provided through a screen reader only class. For Spanish forms, this string is written in Spanish, but read in an English accent because the parent `a` does not have a Spanish `lang` attribute.

## Acceptance Criteria

- [ ] Spanish content within the "Go to tool" link is read in a Spanish accent

## Steps to Recreate

* https://staging.va.gov/find-forms/?q=10-10ez 
* Inspect the "Go to tool" link (styled as a button)

## Proposed Solution

There are several different methods to resolve this ticket.

### Add `lang="es"` to the `a`, and translate "Go to tool" to Spanish
Unfortunately, adding `lang="es"` to the child `<span>` that houses the screen reader only form name won't do the trick when its parent element is a `a` on VO.

[You can test this using this codepen.]()

As a result, we'll need to place the `lang="es"` attribute to the parent `a`, and also translate the "Go to tool" to Spanish so the full link is read out in the same language.

#### Before
```html
<a class="usa-button usa-button-secondary vads-u-margin-bottom--3" 
  href="https://www.va.gov/health-care/apply/application/introduction?_ga=2.18767614.659014729.1618930885-682538563.1615219873">
  Go to online tool 
  <span class="vads-u-visibility--screen-reader">
    for 10-10EZ (esp) Instrucciones y Solicitud de Beneficios Medicos
  </span>
</a>
```

#### After
```html
<!-- Add lang="es" -->
<a lang="es" class="usa-button usa-button-secondary vads-u-margin-bottom--3" 
  href="https://www.va.gov/health-care/apply/application/introduction?_ga=2.18767614.659014729.1618930885-682538563.1615219873">
  <!-- Translate this -->
  Ir al formulario en línea
  <!-- Translate "for" to "por" -->
  <span class="vads-u-visibility--screen-reader">
    por 10-10EZ Instrucciones y Solicitud de Beneficios Medicos
  </span>
</a>
```

### Remove the "Go to online tool" link
Another way to solve this is by removing the "Go to online tool" link. For screen reader users, it's verbose to have two links that lead to the same page and have the same call to action be repeated in close proximity. It's even more verbose when this pattern is repeated more than once in the search results.

For example, a screen reader user may hear upon tabbing:

1. Link, 10-10EZ (esp) Instrucciones y Solicitud de Beneficios Medicos 
2. Link, download VA form 10-10EZ (esp), pdf
3. Link, Go to online tool for, 10-10EZ (esp) Instrucciones y Solicitud de Beneficios Medicos 

This could be simplified by just having two links:
1. Link, 10-10EZ (esp) Instrucciones y Solicitud de Beneficios Medicos 
2. Link, download VA form 10-10EZ (esp), pdf

That being said, this may be a significant design change, and should receive validation from a designer first before proceeding.

## WCAG or Vendor Guidance (optional)

* [Language of Parts: Understanding SC 3.1.2](https://www.w3.org/TR/UNDERSTANDING-WCAG20/meaning-other-lang-id.html)
* [VSP documentation on accessibility for localization](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/resources/localization/accessibility-for-localization.md)

## Screenshots or Trace Logs
<img width="1234" alt="Screen Shot 2021-04-22 at 2 32 47 PM" src="https://user-images.githubusercontent.com/14154792/115767977-b4854280-a377-11eb-9e02-8ea748ea4454.png">

**Comment by MarciMcGuireGCIO** [comment]
@zacharymorel should we pull this guy into the current Sprint, or is it too large/have blockers?

**Comment by zacharymorel** [comment]
It's got blockers I believe. It might change as well with the redesign.

**Comment by MarciMcGuireGCIO** [comment]
Got it.  Thanks!

**Comment by zacharymorel** [comment]
@MarciMcGuireGCIO @johnhashva 
Here's a ticket in the backlog that pertains to the **Go to Online Tool** button text needing to be translated prior to the redesign.

**Comment by MarciMcGuireGCIO** [comment]
@zacharymorel that's a good find.  Wonder what bumped it from a 2 to a launch blocker?

**Comment by zacharymorel** [comment]
The Form Title here wasn't an issue as well. The bold-ing of the text after we removed the URL became an issue because I did not code to allow the lang code.

**Comment by MarciMcGuireGCIO** [comment]
Sounds like we need to meet with Josh and Angela (and possibly Trevor) to decide what our best course of action is.

**Comment by zacharymorel** [comment]
> Sounds like we need to meet with Josh and Angela (and possibly Trevor) to decide what our best course of action is.

The Form Title here wasn't an issue as well. The bold-ing of the text after we removed the URL became an issue because I did not code to allow the lang code.

We have the interpreted Spanish here btw that I can use.

**Comment by MarciMcGuireGCIO** [comment]
Blocked until we have Spanish translation for the updated Go To Online Tool verbiage.

**Comment by zacharymorel** [comment]
@joshkimux when you get the chance, can you read through the the results from https://staging.va.gov/find-forms/?q=10-10ez and tell me if it the current version works out? 

![Screen Shot 2021-09-01 at 10 20 31 AM](https://user-images.githubusercontent.com/26075258/131688191-cb8662d1-51b5-499b-8bc9-6d7f6fe847b9.png)

**Comment by joshkimux** [comment]
![capture-10](https://user-images.githubusercontent.com/14154792/131717038-f4aacd59-a13f-4a24-ae56-9d755b69ded4.PNG)
Awesome! The action link now has `lang="es"` which is what we want! That being said, it looks like the heading "Instrucciones..." above it does not have the `lang="es"` attribute, so it is still read out in an English accent. Are we able to fix that too? ⭐

**Comment by zacharymorel** [comment]
@joshkimux Can you give this another review on staging?

**Comment by joshkimux** [comment]
Yes sir! On it now

**Comment by joshkimux** [comment]
This has been fixed. Stellar work as always @zacharymorel 🥳

**Comment by MarciMcGuireGCIO** [comment]
Thank you again to the dynamic duo! 🥳
